src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/CopyCsvFilesFunctionTests.cs

```diff
@@ -0,0 +1,127 @@
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using File = System.IO.File;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests.Functions;
+
+public class CopyCsvFilesFunctionTests(ProcessorFunctionsIntegrationTestFixture fixture)
+    : ProcessorFunctionsIntegrationTest(fixture)
+{
+    public class CopyCsvFilesTests(ProcessorFunctionsIntegrationTestFixture fixture)
+        : CopyCsvFilesFunctionTests(fixture)
+    {
+        private const string DataFileContent =
+            "time_period,time_identifier,geographic_level,country_code,country_name,enrolments";
+
+        private const string MetadataFileContent =
+            "col_name,col_type,label,indicator_grouping,indicator_unit,indicator_dp,filter_hint,filter_grouping_column";
+
+        [Fact]
+        public async Task Success()
+        {
+            var instanceId = Guid.NewGuid();
+
+            var subjectId = Guid.NewGuid();
+
+            ReleaseVersion releaseVersion = DataFixture.DefaultReleaseVersion()
+                .WithRelease(DataFixture.DefaultRelease());
+
+            var (releaseDataFile, releaseMetaFile) = DataFixture.DefaultReleaseFile()
+                .WithReleaseVersion(releaseVersion)
+                .WithFiles(DataFixture.DefaultFile()
+                    .WithSubjectId(subjectId)
+                    .ForIndex(0, s => s.SetType(FileType.Data))
+                    .ForIndex(1, s => s.SetType(FileType.Metadata))
+                    .GenerateList())
+                .Generate(2)
+                .ToTuple2();
+
+            await AddTestData<ContentDbContext>(context =>
+            {
+                context.ReleaseVersions.Add(releaseVersion);
+                context.ReleaseFiles.AddRange(releaseDataFile, releaseMetaFile);
+            });
+
+            DataSet dataSet = DataFixture.DefaultDataSet();
+
+            await AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion()
+                .WithDataSet(dataSet)
+                .WithStatus(DataSetVersionStatus.Processing)
+                .WithReleaseFileId(releaseDataFile.Id)
+                .WithImports(() => DataFixture
+                    .DefaultDataSetVersionImport()
+                    .WithInstanceId(instanceId)
+                    .WithStage(DataSetVersionImportStage.Pending)
+                    .Generate(1))
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var blobStorageService = GetRequiredService<IPrivateBlobStorageService>();
```

Comment:
Is this a dummy implementation that writes to the filesystem, or using a test container for Azurite, or do we need storage running when running these tests?

Reply:
Aha, I've just found:

```
<PackageReference Include="Testcontainers.Azurite" Version="3.8.0" />
```

:slightly_smiling_face: